src/KKTSolver/krylov.jl

```diff
@@ -410,12 +410,13 @@ function solve!(
 
     # Setup
     d = one(Tv) ./ (kkt.θ .+ kkt.regP)
-    D = Diagonal(d)
+    N = Diagonal(d)
+
+    M = Diagonal(one(Tv) ./ kkt.regD)
 
     # Solve augmented system
-    # Currently assumes that Rd is a multiple of the identity matrix
     _dy, _dx, stats = kkt.f(kkt.A, ξp, ξd,
-        N=D, τ = kkt.regD[1],
+        M=M, N=N,
         atol=kkt.atol, rtol=kkt.rtol
     )
```

Comment:
@amontoison can you confirm that the correspondence (w.r.t `M`, `N` and right-hand sides) is correct here?

Reply:
The preconditioner is `[M 0; 0 N]` now and not `[M 0; 0 I]` anymore.
If you want to solve
```julia
[E  0] [x] = [ b ]
[0 -F] [y]   [ c ]
```
you need `M = inv(E)` and `N = inv(F)`.

Examples:
https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/master/examples/test_tricg.jl
https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/master/examples/test_trimr.jl

Reply:
I want to solve
```
[-(Θ⁻¹+Rp)  A' ] [dx] = [ ξd ]
[    A      Rd ] [dy]   [ ξp ]
```
Writing it in the trimr default form yields
```
[ Rd       A   ] [dy] = [ ξp ]
[ A'  -(Θ⁻¹+Rp)] [dx]   [ ξd ]
```
i.e., `E = Rd` and `F = (Θ⁻¹+Rp)`, so that
```
M = inv(E) = Rd⁻¹
N = inv(F) = (Θ⁻¹ + Rp)⁻¹
```

I think I wrote it wrong the first time (`τ` should have been negative) 😅 
For some reason it was still working though

Reply:
Yes, it's right. FYI, you can use the kwarg `flip=true` to solve
```julia
[-M   A] [x] = [b]
[ A'  N] [y]   [c]
```
instead of
```julia
[M    A] [x] = [b]
[A'  -N] [y]   [c]
```